src/store/metadata/oxigraph/store.py

```diff
@@ -344,7 +348,46 @@ def get_sub_topics(self, topic_id: str) -> Optional[Dict]:  # pragma: no cover
         """
         Get all sub-topics for a specific topic
         """
-        raise NotImplementedError
+
+        # Populate the graph from the database
```

Comment:
there are times in every deveopers career where you feel the need to kick yourself hard with a big boot 😄 

your code works (and well figured out btw) ....but... you can probably just do this sorta thing:

```
    def get_sub_topics(self, topic_id: str) -> Optional[Dict]:  # pragma: no cover
        """
        Get all sub-topics for a specific topic
        """

        all_topics = self.get_topics()
        topics_with_parents = [topic for topic in all_topics["topics"] if topic.get("parent_topics", None) is not None]
        sub_topics = [
            topic for topic in topics_with_parents if any([x.endswith(topic_id) for x in topic["parent_topics"]]) 
        ]
        
        if len(sub_topics) == 0:
            return None

        return {
            "@context": "https://staging.idpd.uk/ns#",
            "@id": f"https://staging.idpd.uk/topics/{topic_id}/subtopics",
            "@type": "hydra:Collection",
            "topics": sub_topics,
            "count": {len(sub_topics)},
            "offset": 0
            }
```

Reply:
Thanks, I knew there was an easier way to do it *tunnel vision from looking at it for too long*